var documenterSearchIndex = {"docs":
[{"location":"create_db/#Create-a-new-Clasnip-database","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"","category":"section"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"Clasnip performs sample classification based on specialized databases. The Clasnip database is constructed with input samples with group labels, and one of the samples is marked as a reference. An input sample is a nucleotide sequence, or a collection of nucleotide sequences in the FASTA format. A sample can have sequences of a single gene, multiple genes, contigs, or even a genome. Users do not need to create any alignment file before submitting.","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"To create a database, you need to sign up (it is free.), and log in to your account. Then, select Menu (top bar) -> Create Database.","category":"page"},{"location":"create_db/#Database-information","page":"Create a new Clasnip database","title":"Database information","text":"","category":"section"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"To create a Clasnip database, users need to provide the following information:","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"Database ID: It is a unique name to the database. Avoid special characters. (Image: )\nTaxonomic rank: Select one of the items: genus, species, strain.\nTaxonomic name (scientific name): The official scientific name describing all sequences in the database.\nCommon name: Optional.\nDatabase type: Select one from the items:\ngenomic: It is a genomic database. Samples in the database contain genomic sequences, but some samples may be genes or fragments.\ngenomic - all samples are assemblies: It is a genomic database, and all samples are whole genome sequences (contigs, scaffolds, or complete genomes).\nmultiple genes: The database contain samples with more than one genes.\nsingle gene: The database contain samples with only one gene.\nCovered regions or genes: If database type is multiple genes or single gene, you need to enter what regions or genes are included in the database.\nDatabase file: See below. (Image: )","category":"page"},{"location":"create_db/#Database-file","page":"Create a new Clasnip database","title":"Database file","text":"","category":"section"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"You need to upload a compressed database folder to Clasnip. The folder shall contain several sub-folders holding sequences in FASTA format. Each sub-folder is considered as a group, and each FASTA file is considered as a sample.","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"details: What is the FASTA format?\nA sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line (defline) is distinguished from the sequence data by a greater-than (\">\") symbol at the beginning. An example sequence in FASTA format is:>Name1 description\nGTTGATGGGGTCATTTGAGTTTATGTTAAGGGCCCATAG\nCTCAGGCGGTTAGAGTGCACCC\n>Name2 description\nATAGTTTTTTTGTTCTAGGGATTTTTTTTTAGAGCAATA","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"The tree view of the compressed folder looks like this:","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"Pectobacteria_database.zip\n├── Pectobacterium_actinidiae\n│   ├── GCA_000803315.1_ASM80331v1_genomic.fasta\n│   ├── GCA_007833835.1_ASM783383v1_genomic.fasta\n│   ├── GCA_017426805.1_Pca_genomic.fasta\n│   └── GCA_024343335.1_ASM2434333v1_genomic.fasta\n├── Pectobacterium_aquaticum\n│   ├── GCA_002869945.1_ASM286994v1_genomic.fasta\n│   ├── GCA_003382565.3_ASM338256v3_genomic.fasta\n├── Pectobacterium_aroidearum\n│   ├── GCA_000023605.1_ASM2360v1_genomic.fasta\n│   ├── GCA_013449715.1_ASM1344971v1_genomic.fasta\n│   ├── GCA_013913605.1_ASM1391360v1_genomic.fasta","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"In the example, the database contain three groups, named Pectobacterium_actinidiae, Pectobacterium_aquaticum, and Pectobacterium_aroidearum . Those groups have 4, 2 and 3 samples,respectively.","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"info: Specs of a database file\nAccepted compressed methods for database file: zip, tar.gz, tar.bz2, tar.xz, tar.Z\nAllowed FASTA file extensions: fasta, fa, fna, fs. FASTA should not be compressed. \nNested folders are allowed: Group_A/Group_A1/Sample is valid, and the group of Sample is automatically converted to Group_A/Group_A1\nDatabase size limit: 200M, or shown in Clasnip.com.","category":"page"},{"location":"create_db/#Reference-sequence","page":"Create a new Clasnip database","title":"Reference sequence","text":"","category":"section"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"After uploading the database file, user need to select a reference sequence from the FASTA files. The reference should allow all other sequences to be mapped to. If any sequence fails to map to the reference, it will not be analyzed. If a sample barely maps to the reference, the classification performance will be compromised.","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"(Image: )","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"If the samples are too diverse to be mapped using a single reference, we can","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"Click \"Build pangenome\". This function is useful when no reference sequence can represent all groups in the database. You can choose one reference sequence, and a pangenome will be built based on the sequence. The unmapped region of other sequences will be appended to the pangenome, and the pangenome will be the reference genome.\nIf the previous method does not generate good mapping results, you can also manually create a pangenome, and added the pangenome as a normal FASTA file in the database. Select this as reference, and click \"Exclude reference in classification\". This option is useful when the reference is artificial and only used for a better alignment.","category":"page"},{"location":"create_db/#Submit","page":"Create a new Clasnip database","title":"Submit","text":"","category":"section"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"Congratualtions! You are all set. Now you can click \"SUBMIT\" to create the database. It may take some time depending on sequence lengths. You can go to the report page to see whether the task is finished.","category":"page"},{"location":"create_db/","page":"Create a new Clasnip database","title":"Create a new Clasnip database","text":"After the task is finished, you can review the detailed database information by selecting Menu (top bar) -> Database Information.","category":"page"},{"location":"db_info/#Clasnip-Database-Information","page":"Database Information","title":"Clasnip Database Information","text":"","category":"section"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"Clasnip performs sample classification based on specialized databases. When creating a database, Clasnip automatically analyzes performance based on the samples in the database. The database information and performance reports are accessible by all users. ","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"To browse database information and performance reports, go to Menu (top bar) -> Database Information -> **Choose A Clasnip Database **.","category":"page"},{"location":"db_info/#Database-information","page":"Database Information","title":"Database information","text":"","category":"section"},{"location":"db_info/#Metadata","page":"Database Information","title":"Metadata","text":"","category":"section"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"Database metadata includes:","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"Database ID: The unique ID of clasnip database.\nTaxonomy: Ccientific name, and taxonomy level.\nGroup By: What are the samples grouped by? Eg: species, stains, haplotypes.\nRegion: What region/genes are covered?\nReference Sequence: The reference sequence used for mapping. If it contains .pan., it is a pan-genome containing not only the reference sequence, but also unmapped regions from other samples. \nCreated By: A user who creates the database.\nDate Created: Date in the format of YYYY-MM-DD.","category":"page"},{"location":"db_info/#Classification-groups","page":"Database Information","title":"Classification groups","text":"","category":"section"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"This section lists classification groups in the database.","category":"page"},{"location":"db_info/#Accuracy","page":"Database Information","title":"Accuracy","text":"","category":"section"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"The percent of accuracy computed based on the samples in the database.","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"Accuracy = (TP + TN)  (TP + TN + FP + FN)","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"where","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"TP (True Positive) = Group X sample correctly identified as X.\nTN (True Negative) = Non group X sample correctly identified as not X.\nFP (False Positive) = Non group X sample incorrectly identified as X.\nFN (False Negative) = Group X sample incorrectly identified as not X.","category":"page"},{"location":"db_info/#Detailed-classification-performance","page":"Database Information","title":"Detailed classification performance","text":"","category":"section"},{"location":"db_info/#Heatmaps-and-dendrograms","page":"Database Information","title":"Heatmaps and dendrograms","text":"","category":"section"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"Several plots were generated using different data in the forms of heatmap and dendrogram. Plots may automatically hided, you need to click \"Dendrogram\", or the sub-buttons under Heatmap to view the plot. ","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"The plots are interactive, and you can","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"resize the plot by dragging the bottom-right corner of the canvas.\nhover on heatmap cells to show the description of X and Y axises, and numbers. ","category":"page"},{"location":"db_info/","page":"Database Information","title":"Database Information","text":"warn: Warn\nSome plots might be very big and take a while to show and resize.","category":"page"},{"location":"db_info/#Group-Similarity-Based-on-Sample-Identities","page":"Database Information","title":"Group Similarity Based on Sample Identities","text":"","category":"section"},{"location":"algorithm/#Algorithm","page":"Algorithm","title":"Algorithm","text":"","category":"section"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"Clasnip is a fast, sound and reliable bioinformatics classifier that can facilitate the detection of pathogens, and it is important for disease containment and cost reductions for quick reaction in case of an outbreak or invasion of an exotic plant disease. ","category":"page"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"Clasnip was programmed to classify DNA sequences of closely-related microorganisms, and generate an SNP table with a pre-processed database. ","category":"page"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"The framework of Clasnip is illustrated as below. The back-end (server) and the front-end (web interface) are two separated programs. All classification tasks are done at the server side, including database building and new sample classification. The user-interface is only used for display purposes.","category":"page"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"(Image: )","category":"page"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"Please read our paper for details [1].","category":"page"},{"location":"algorithm/","page":"Algorithm","title":"Algorithm","text":"Chuan J, Xu H, Hammill DL, Hale L, Chen W, Li X. 2023. Clasnip: a web-based intraspecies classifier and multi-locus sequence typing for pathogenic microorganisms using fragmented sequences. PeerJ 11:e14490 https://doi.org/10.7717/peerj.14490","category":"page"},{"location":"changelog/#Change-Log","page":"Change Log","title":"Change Log","text":"","category":"section"},{"location":"changelog/#v2.5.0-**DB-NEED-RE-BUILD**","page":"Change Log","title":"v2.5.0 - DB NEED RE-BUILD","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: heatmap/phylogenetic tree for samples (db samples and new samples).\nOptim: stores some data to supports the feature above.\nChange: retain more types of files after creating db.\nOther optimization and updates.\nChange: lock types.\nFix: closing db files after loading.\nNew: functions.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"API:","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Fix: reports: db concatenate does not work for number of db > 2.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"UI:","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Change: all heatmaps with Plotly heatmaps and dendrograms.\nChange: reports can have heatmaps and dendrograms.","category":"page"},{"location":"changelog/#v2.4.0-**DB-NEED-RE-BUILD**","page":"Change Log","title":"v2.4.0 - DB NEED RE-BUILD","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Algorithm: set uppercase to REF and ALT. This might change db and results.\nOptim: Speed gain by updating JobSchedulers.","category":"page"},{"location":"changelog/#v2.3.1","page":"Change Log","title":"v2.3.1","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"New: script.\nFeat: script.\nCompat: JobSchedulers.\nAlgorithm: Load identity distributions when loading database.\nFeat: Lock type changes.","category":"page"},{"location":"changelog/#v2.3.0","page":"Change Log","title":"v2.3.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feat: Now some API accepts more morden token validation.","category":"page"},{"location":"changelog/#v2.2.0","page":"Change Log","title":"v2.2.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Algorithm: Probability update: Probability considers P(not classified to any groups).   It assumes the sample can only be classified to one group.  \nTherefore, if sumCDF <= 1,   P(not classified to any groups) = 1 - sumCDF,   and the real P is CDF.  \nIf sumCDF > 1,   P(not classified to any groups) = 0,   real P = CDF/sumCDF (the same as before).\nFeature: colorful repl http messages.","category":"page"},{"location":"changelog/#v2.1.4","page":"Change Log","title":"v2.1.4","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: allow db file containing multiple depth of group: GRO/OP/SAMPLE.\nFeature: handling failed dbs.\nFix: restart server into forever loop.","category":"page"},{"location":"changelog/#v2.1.3","page":"Change Log","title":"v2.1.3","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: ClasnipMux enables Package's precompilation at some point.\nOptimize: faster API response.\nRemove: Dependency HttpCommon.jl.\nFix: DB generation: re-write using JobSchedulers.","category":"page"},{"location":"changelog/#v2.1.2","page":"Change Log","title":"v2.1.2","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Fix: db reanalyzing script: building pangenome error.","category":"page"},{"location":"changelog/#v2.1.1","page":"Change Log","title":"v2.1.1","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: better type assertion/consistency for better precompiled code (>10X faster for computing identity and 5X faster for computing SNP heatmap in db build).\nOptimize: some computational intensive loops.","category":"page"},{"location":"changelog/#v2.1.0","page":"Change Log","title":"v2.1.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: Save time and RAM for new sample classification: do not parse ALT2PROBs to human readable group columns for MLST tables with less than 100_000 rows. If number of rows is larger , the tabular MLST file will not be generated, but MLST data will be stored in a different format. \nOptimize: Re-write a multi-threaded version of DB generation. Default is 4 threaded. This version has 0.028% (244/865931) differences in ALT2PROBs in test data. Checked the first two records, new verison is correct. \nFeature: DB storing updates.","category":"page"},{"location":"changelog/#v2.0.0","page":"Change Log","title":"v2.0.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: New Clasnip DB type was created to store in a special format. \nThe size of all 14 Clasnip DB files reduced 18.3X, from 5520MB to 302MB.\nDB loading time is 4.9X faster.\nDatabase generation time is also reduced because of the time of saving DB.\nMigration script is avaiable.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Database Compression Ratio DB Loading Speed Ratio File Size, Old (MB) File Size, New (MB) Load Time, Old (s) Convert Time (s) Save Time, New (s) Load Time, New (s) Uncompress Time, New (s)\navipoxvirus_c1_3_0 40.7 2.7 31.85 0.78 1.70 1.95 1.18 0.16 0.48\nClavibacter_202301_c1_3_0 20.3 4.9 703.58 34.68 7.26 3.37 0.09 0.03 1.45\nclso_v6_16_23s_c1_3_0 1.5 1.3 0.06 0.04 0.00 0.03 0.00 0.00 0.00\nclso_v6_16s_c1_3_0 1.3 1.4 0.06 0.04 1.28 1.24 1.57 0.41 0.53\nclso_v6_50s_c1_3_0 1.7 1.1 0.08 0.04 0.00 0.08 0.00 0.00 0.00\nclso_v6_c1_3_0 41.7 3.2 34.47 0.83 2.04 1.38 1.60 0.38 0.26\nDickeya_202210_c1_3_0 19.1 40.1 1,079.00 56.35 16.39 8.70 0.05 0.01 0.40\nPectobacterium_202210_c1_3_0 15.8 2.7 1,278.91 81.07 14.91 4.92 0.06 1.09 4.38\nPectobacterium_202407_c1_3_0 15.5 4.2 1,369.15 88.18 16.54 9.01 0.06 1.03 2.88\npotato_virus_Y_phylogroups_quarantine_c1_3_0 22.1 7.8 1.85 0.08 0.02 0.04 0.04 0.00 0.00\nRalstonia_solanacearum_complex_20230515_c1_3_0 25.2 7.9 908.75 36.13 12.40 4.20 0.05 0.01 1.56\nRust_fungi_ITS2_20230315_c1_3_0 10.9 2.7 1.88 0.17 0.01 0.02 0.00 0.00 0.00\nRust_fungi_ITS2_20240412_c1_3_0 10.9 2.8 1.88 0.17 0.01 0.03 0.02 0.00 0.00\nXylella_fastidiosa_20230301_c1_3_0 36.0 82.0 108.00 3.00 1.98 6.32 0.01 0.00 0.02\nAll 18.3 4.9 5,519.50 301.58 74.53 41.28 4.73 3.14 11.98","category":"page"},{"location":"changelog/#v1.3.0","page":"Change Log","title":"v1.3.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: algorithm. This requres recalculation of existing databases.\nOptimize: heatmap figure size now adjusted to both numbers of characters and groups.","category":"page"},{"location":"changelog/#Summary-(v0.8-v1.2)","page":"Change Log","title":"Summary (v0.8 -> v1.2)","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"The new sample classification module in Clasnip is very fast because the variant database is precomputed and can be preloaded into memory. However, the pre-computation of variant database during database building required a lot of time and CPU/memory resources.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"I profiled each step of database building, and optimized the time-limited steps, as well as tools and programming algorithm. ","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"More accurate and 10X faster for mapping and variant calling steps. This change has drastically improved classification performance for Candidatus Liberibacter solanacearum.\nI optimized the data structure and algorithm of variant databases. The new version has achieved 1.4X faster database computation speed and 42.8% less storage size.\nCorrected a bug: the SNP coverage in the diagonal cells of SNP heatmap was doubled.\nWhen building a database, users can choose to create a pangenome using all sequences.\nOther improvements.","category":"page"},{"location":"changelog/#v1.2.0","page":"Change Log","title":"v1.2.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: The total size of 12 databases shrinked to 57.2% comparing to v1.0.0. New sample analysis will be faster theoritically. DB accuracy is the same as v1.0.0.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Name Old Size New Size Change\navipoxvirus_c1_2_0 53570014 33166727 0.619129\nClavibacter_202301_c1_2_0 1255558018 719021314 0.572671\nclso_v6_16_23s_c1_2_0 79004 65844 0.833426\nclso_v6_16s_c1_2_0 68876 59836 0.86875\nclso_v6_50s_c1_2_0 102956 80204 0.779012\nclso_v6_c1_2_0 60107291 36024803 0.599342\nDickeya_202210_c1_2_0 1982661237 1131665789 0.570781\nPectobacterium_202210_c1_2_0 2408675601 1341280377 0.556854\npotato_virus_Y_phylogroups_quarantine_c1_2_0 3219547 1935059 0.601035\nRalstonia_solanacearum_complex_20230515_c1_2_0 1510960255 886691638 0.58684\nRust_fungi_ITS2_20230315_c1_2_0 3596289 1966430 0.546794\nXylella_fastidiosa_20230301_c1_2_0 180763273 111769652 0.618321\nTotal 7459362361 4263727673 0.571594","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: db generation: run time reduced to 72% comparing to v1.0.0. \nFix: group coverage is no longer doubled when group in X axis == group in y axis. This affects all figures named 'Group Similarity Based on SNP Differences (For Single-Gene Database).'\nOther improvements.","category":"page"},{"location":"changelog/#v1.1.0","page":"Change Log","title":"v1.1.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Compat: JobScheduler v0.9.","category":"page"},{"location":"changelog/#v0.8.0-v1.0.0","page":"Change Log","title":"v0.8.0 -> v1.0.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: database performance improvement. Following stats are overall accuracy (%).\nCandidatus Liberibacter solanacearum (genomic): 99.12 -> 100.00\nCandidatus Liberibacter solanacearum (16S): 97.37 -> 98.68\nAll other databases created prior to 2024-Mar-08: 100.00 -> 100.00\nFeature: allow users create pangenome using a reference sequence.\nOptimize: change bioinformatics dependencies and related algorithms to reduce storage and increase speed.\nOther minor improvements.","category":"page"},{"location":"manual/#Manual","page":"Manual","title":"Manual","text":"","category":"section"},{"location":"#Clasnip-Documentation","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"","category":"section"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"Clasnip (www.clasnip.com) is an easy-to-use web-based platform for the classification and similarity evaluation of closely related microorganisms at interspecies and intraspecies levels.","category":"page"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"At the current stage, it provides reliable classification for quarantine and regulatory pathogens.","category":"page"},{"location":"#Citation-and-technical-details","page":"Clasnip Documentation","title":"Citation and technical details","text":"","category":"section"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"Chuan J, Xu H, Hammill DL, Hale L, Chen W, Li X. 2023. Clasnip: a web-based intraspecies classifier and multi-locus sequence typing for pathogenic microorganisms using fragmented sequences. PeerJ 11:e14490 https://doi.org/10.7717/peerj.14490","category":"page"},{"location":"#Features","page":"Clasnip Documentation","title":"Features","text":"","category":"section"},{"location":"#Just-input-nucleotide(s)-and-choose-databases.","page":"Clasnip Documentation","title":"Just input nucleotide(s) and choose databases.","text":"","category":"section"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"No matter they are PCR-amplicons, genes, fragmented sequences, contigs, or even whole genomes, as long as they share similarity with databases.\nNo need to align or truncate sequences. (Image: )","category":"page"},{"location":"#Clear-classification-report-with-visualizations","page":"Clasnip Documentation","title":"Clear classification report with visualizations","text":"","category":"section"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"Tabular classification summary (Image: )\nDendrogram and heatmap show the placement of your sample (Image: )\nMLST table showing SNP/INDEL and their prevalence in database (Image: )","category":"page"},{"location":"#Database-of-your-interest-not-found?","page":"Clasnip Documentation","title":"Database of your interest not found?","text":"","category":"section"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"Create one by yourself, and review classification performance in a glance like an expert (login required)","category":"page"},{"location":"","page":"Clasnip Documentation","title":"Clasnip Documentation","text":"Guided database-creating experience (Image: )\nAuto-generated heatmaps and dendrograms (Image: )\nReview performances (TPR / sensitivity, TNR / specificity, etc.) (Image: )","category":"page"}]
}
