var documenterSearchIndex = {"docs":
[{"location":"changelog/#Change-Log","page":"Change Log","title":"Change Log","text":"","category":"section"},{"location":"changelog/#v2.5.0-**DB-NEED-RE-BUILD**","page":"Change Log","title":"v2.5.0 - DB NEED RE-BUILD","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: heatmap/phylogenetic tree for samples (db samples and new samples).\nOptim: stores some data to supports the feature above.\nChange: retain more types of files after creating db.\nOther optimization and updates.\nChange: lock types.\nFix: closing db files after loading.\nNew: functions.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"API:","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Fix: reports: db concatenate does not work for number of db > 2.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"UI:","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Change: all heatmaps with Plotly heatmaps and dendrograms.\nChange: reports can have heatmaps and dendrograms.","category":"page"},{"location":"changelog/#v2.4.0-**DB-NEED-RE-BUILD**","page":"Change Log","title":"v2.4.0 - DB NEED RE-BUILD","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Algorithm: set uppercase to REF and ALT. This might change db and results.\nOptim: Speed gain by updating JobSchedulers.","category":"page"},{"location":"changelog/#v2.3.1","page":"Change Log","title":"v2.3.1","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"New: script.\nFeat: script.\nCompat: JobSchedulers.\nAlgorithm: Load identity distributions when loading database.\nFeat: Lock type changes.","category":"page"},{"location":"changelog/#v2.3.0","page":"Change Log","title":"v2.3.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feat: Now some API accepts more morden token validation.","category":"page"},{"location":"changelog/#v2.2.0","page":"Change Log","title":"v2.2.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Algorithm: Probability update: Probability considers P(not classified to any groups).   It assumes the sample can only be classified to one group.  \nTherefore, if sumCDF <= 1,   P(not classified to any groups) = 1 - sumCDF,   and the real P is CDF.  \nIf sumCDF > 1,   P(not classified to any groups) = 0,   real P = CDF/sumCDF (the same as before).\nFeature: colorful repl http messages.","category":"page"},{"location":"changelog/#v2.1.4","page":"Change Log","title":"v2.1.4","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: allow db file containing multiple depth of group: GRO/OP/SAMPLE.\nFeature: handling failed dbs.\nFix: restart server into forever loop.","category":"page"},{"location":"changelog/#v2.1.3","page":"Change Log","title":"v2.1.3","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Feature: ClasnipMux enables Package's precompilation at some point.\nOptimize: faster API response.\nRemove: Dependency HttpCommon.jl.\nFix: DB generation: re-write using JobSchedulers.","category":"page"},{"location":"changelog/#v2.1.2","page":"Change Log","title":"v2.1.2","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Fix: db reanalyzing script: building pangenome error.","category":"page"},{"location":"changelog/#v2.1.1","page":"Change Log","title":"v2.1.1","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: better type assertion/consistency for better precompiled code (>10X faster for computing identity and 5X faster for computing SNP heatmap in db build).\nOptimize: some computational intensive loops.","category":"page"},{"location":"changelog/#v2.1.0","page":"Change Log","title":"v2.1.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: Save time and RAM for new sample classification: do not parse ALT2PROBs to human readable group columns for MLST tables with less than 100_000 rows. If number of rows is larger , the tabular MLST file will not be generated, but MLST data will be stored in a different format. \nOptimize: Re-write a multi-threaded version of DB generation. Default is 4 threaded. This version has 0.028% (244/865931) differences in ALT2PROBs in test data. Checked the first two records, new verison is correct. \nFeature: DB storing updates.","category":"page"},{"location":"changelog/#v2.0.0","page":"Change Log","title":"v2.0.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: New Clasnip DB type was created to store in a special format. \nThe size of all 14 Clasnip DB files reduced 18.3X, from 5520MB to 302MB.\nDB loading time is 4.9X faster.\nDatabase generation time is also reduced because of the time of saving DB.\nMigration script is avaiable.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Database Compression Ratio DB Loading Speed Ratio File Size, Old (MB) File Size, New (MB) Load Time, Old (s) Convert Time (s) Save Time, New (s) Load Time, New (s) Uncompress Time, New (s)\navipoxvirusc13_0 40.7 2.7 31.85 0.78 1.70 1.95 1.18 0.16 0.48\nClavibacter202301c130 20.3 4.9 703.58 34.68 7.26 3.37 0.09 0.03 1.45\nclsov61623sc130 1.5 1.3 0.06 0.04 0.00 0.03 0.00 0.00 0.00\nclsov616sc13_0 1.3 1.4 0.06 0.04 1.28 1.24 1.57 0.41 0.53\nclsov650sc13_0 1.7 1.1 0.08 0.04 0.00 0.08 0.00 0.00 0.00\nclsov6c130 41.7 3.2 34.47 0.83 2.04 1.38 1.60 0.38 0.26\nDickeya202210c130 19.1 40.1 1,079.00 56.35 16.39 8.70 0.05 0.01 0.40\nPectobacterium202210c130 15.8 2.7 1,278.91 81.07 14.91 4.92 0.06 1.09 4.38\nPectobacterium202407c130 15.5 4.2 1,369.15 88.18 16.54 9.01 0.06 1.03 2.88\npotatovirusYphylogroupsquarantinec13_0 22.1 7.8 1.85 0.08 0.02 0.04 0.04 0.00 0.00\nRalstoniasolanacearumcomplex20230515c130 25.2 7.9 908.75 36.13 12.40 4.20 0.05 0.01 1.56\nRustfungiITS220230315c130 10.9 2.7 1.88 0.17 0.01 0.02 0.00 0.00 0.00\nRustfungiITS220240412c130 10.9 2.8 1.88 0.17 0.01 0.03 0.02 0.00 0.00\nXylellafastidiosa20230301c13_0 36.0 82.0 108.00 3.00 1.98 6.32 0.01 0.00 0.02\nAll 18.3 4.9 5,519.50 301.58 74.53 41.28 4.73 3.14 11.98","category":"page"},{"location":"changelog/#v1.3.0","page":"Change Log","title":"v1.3.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: algorithm. This requres recalculation of existing databases.\nOptimize: heatmap figure size now adjusted to both numbers of characters and groups.","category":"page"},{"location":"changelog/#Summary-(v0.8-v1.2)","page":"Change Log","title":"Summary (v0.8 -> v1.2)","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"The new sample classification module in Clasnip is very fast because the variant database is precomputed and can be preloaded into memory. However, the pre-computation of variant database during database building required a lot of time and CPU/memory resources.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"I profiled each step of database building, and optimized the time-limited steps, as well as tools and programming algorithm. ","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"More accurate and 10X faster for mapping and variant calling steps. This change has drastically improved classification performance for Candidatus Liberibacter solanacearum.\nI optimized the data structure and algorithm of variant databases. The new version has achieved 1.4X faster database computation speed and 42.8% less storage size.\nCorrected a bug: the SNP coverage in the diagonal cells of SNP heatmap was doubled.\nWhen building a database, users can choose to create a pangenome using all sequences.\nOther improvements.","category":"page"},{"location":"changelog/#v1.2.0","page":"Change Log","title":"v1.2.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: The total size of 12 databases shrinked to 57.2% comparing to v1.0.0. New sample analysis will be faster theoritically. DB accuracy is the same as v1.0.0.","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"| Name                                           | Old Size   | New Size   | Change   |   |:-|-:|-:|:-| | avipoxvirusc120                             | 53570014   | 33166727   | 0.619129 |   | Clavibacter202301c120                      | 1255558018 | 719021314  | 0.572671 |   | clsov61623sc120                          | 79004      | 65844      | 0.833426 |   | clsov616sc120                             | 68876      | 59836      | 0.86875  |   | clsov650sc120                             | 102956     | 80204      | 0.779012 |   | clsov6c120                                 | 60107291   | 36024803   | 0.599342 |   | Dickeya202210c120                          | 1982661237 | 1131665789 | 0.570781 |   | Pectobacterium202210c120                   | 2408675601 | 1341280377 | 0.556854 |   | potatovirusYphylogroupsquarantinec120   | 3219547    | 1935059    | 0.601035 |   | Ralstoniasolanacearumcomplex20230515c120 | 1510960255 | 886691638  | 0.58684  |   | RustfungiITS220230315c120                | 3596289    | 1966430    | 0.546794 |   | Xylellafastidiosa20230301c12_0             | 180763273  | 111769652  | 0.618321 |   | Total                                      | 7459362361 | 4263727673 | 0.571594 |  ","category":"page"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: db generation: run time reduced to 72% comparing to v1.0.0. \nFix: group coverage is no longer doubled when group in X axis == group in y axis. This affects all figures named 'Group Similarity Based on SNP Differences (For Single-Gene Database).'\nOther improvements.","category":"page"},{"location":"changelog/#v1.1.0","page":"Change Log","title":"v1.1.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Compat: JobScheduler v0.9.","category":"page"},{"location":"changelog/#v0.8.0-v1.0.0","page":"Change Log","title":"v0.8.0 -> v1.0.0","text":"","category":"section"},{"location":"changelog/","page":"Change Log","title":"Change Log","text":"Optimize: database performance improvement. Following stats are overall accuracy (%).\nCandidatus Liberibacter solanacearum (genomic): 99.12 -> 100.00\nCandidatus Liberibacter solanacearum (16S): 97.37 -> 98.68\nAll other databases created prior to 2024-Mar-08: 100.00 -> 100.00\nFeature: allow users create pangenome using a reference sequence.\nOptimize: change bioinformatics dependencies and related algorithms to reduce storage and increase speed.\nOther minor improvements.","category":"page"},{"location":"manual/#Manual","page":"Manual","title":"Manual","text":"","category":"section"},{"location":"#Clasnip-Documentation","page":"Home","title":"Clasnip Documentation","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Clasnip (www.clasnip.com) is an easy-to-use web-based platform for the classification and similarity evaluation of closely related microorganisms at interspecies and intraspecies levels.","category":"page"},{"location":"","page":"Home","title":"Home","text":"At the current stage, it provides reliable classification for quarantine and regulatory pathogens.","category":"page"},{"location":"#Citation-and-technical-details","page":"Home","title":"Citation and technical details","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Chuan J, Xu H, Hammill DL, Hale L, Chen W, Li X. 2023. Clasnip: a web-based intraspecies classifier and multi-locus sequence typing for pathogenic microorganisms using fragmented sequences. PeerJ 11:e14490 https://doi.org/10.7717/peerj.14490","category":"page"},{"location":"#Features","page":"Home","title":"Features","text":"","category":"section"},{"location":"#Just-input-nucleotide(s)-and-choose-databases.","page":"Home","title":"Just input nucleotide(s) and choose databases.","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"No matter they are PCR-amplicons, genes, fragmented sequences, contigs, or even whole genomes, as long as they share similarity with databases.\nNo need to align or truncate sequences. (Image: )","category":"page"},{"location":"#Clear-classification-report-with-visualizations","page":"Home","title":"Clear classification report with visualizations","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Tabular classification summary (Image: )\nDendrogram and heatmap show the placement of your sample (Image: )\nMLST table showing SNP/INDEL and their prevalence in database (Image: )","category":"page"},{"location":"#Database-of-your-interest-not-found?","page":"Home","title":"Database of your interest not found?","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Create one by yourself, and review classification performance in a glance like an expert (login required)","category":"page"},{"location":"","page":"Home","title":"Home","text":"Guided database-creating experience (Image: )\nAuto-generated heatmaps and dendrograms (Image: )\nReview performances (TPR / sensitivity, TNR / specificity, etc.) (Image: )","category":"page"}]
}
