<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Create New Database · Clasnip Documentation</title><meta name="title" content="Create New Database · Clasnip Documentation"/><meta property="og:title" content="Create New Database · Clasnip Documentation"/><meta property="twitter:title" content="Create New Database · Clasnip Documentation"/><meta name="description" content="Documentation for Clasnip Documentation."/><meta property="og:description" content="Documentation for Clasnip Documentation."/><meta property="twitter:description" content="Documentation for Clasnip Documentation."/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../search_index.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">Clasnip Documentation</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../">Clasnip Documentation</a></li><li><a class="tocitem" href="../new_analysis/">New Analysis</a></li><li><a class="tocitem" href="../reports/">Job Reports</a></li><li><a class="tocitem" href="../db_info/">Database Information</a></li><li class="is-active"><a class="tocitem" href>Create New Database</a><ul class="internal"><li><a class="tocitem" href="#Database-information"><span>Database information</span></a></li><li><a class="tocitem" href="#Database-file"><span>Database file</span></a></li><li><a class="tocitem" href="#Reference-sequence"><span>Reference sequence</span></a></li></ul></li><li><a class="tocitem" href="../algorithm/">Algorithm</a></li><li><a class="tocitem" href="../changelog/">Change Log</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Create New Database</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Create New Database</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/cihga39871/Clasnip.com" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/cihga39871/Clasnip.com/blob/main/docs/src/create_db.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="Creating-new-database"><a class="docs-heading-anchor" href="#Creating-new-database">Creating new database</a><a id="Creating-new-database-1"></a><a class="docs-heading-anchor-permalink" href="#Creating-new-database" title="Permalink"></a></h1><p>Clasnip performs sample classification based on specialized databases. The Clasnip database is constructed with input samples with group labels, and one of the samples is marked as a reference. An input sample is a nucleotide sequence, or a collection of nucleotide sequences in the FASTA format. A sample can have sequences of a single gene, multiple genes, contigs, or even a genome. Users do not need to create any alignment file before submitting.</p><p>To create a database, you need to <strong>sign up</strong> (it is free.), and <strong>log in</strong> to your account. Then, select <strong>Menu</strong> (top bar) -&gt; <strong>Create Database</strong>.</p><h2 id="Database-information"><a class="docs-heading-anchor" href="#Database-information">Database information</a><a id="Database-information-1"></a><a class="docs-heading-anchor-permalink" href="#Database-information" title="Permalink"></a></h2><p>To create a Clasnip database, users need to provide the following information:</p><ul><li><strong>Database ID</strong>: It is a unique name to the database. Avoid special characters. <img src="../assets/create_db1.png" alt/></li><li><strong>Taxonomic rank</strong>: Select one of the items: genus, species, strain.</li><li><strong>Taxonomic name (scientific name)</strong>: The official scientific name describing all sequences in the database.</li><li><strong>Common name</strong>: Optional.</li><li><strong>Database type</strong>: Select one from the items:<ul><li><strong>genomic</strong>: It is a genomic database. Samples in the database contain genomic sequences, but some samples may be genes or fragments.</li><li><strong>genomic - all samples are assemblies</strong>: It is a genomic database, and all samples are whole genome sequences (contigs, scaffolds, or complete genomes).</li><li><strong>multiple genes</strong>: The database contain samples with more than one genes.</li><li><strong>single gene</strong>: The database contain samples with only one gene.</li></ul></li><li><strong>Covered regions or genes</strong>: If database type is multiple genes or single gene, you need to enter what regions or genes are included in the database.</li><li><strong>Database file</strong>: See below. <img src="../assets/create_db2.png" alt/></li></ul><h2 id="Database-file"><a class="docs-heading-anchor" href="#Database-file">Database file</a><a id="Database-file-1"></a><a class="docs-heading-anchor-permalink" href="#Database-file" title="Permalink"></a></h2><p>You need to upload a <strong>compressed</strong> database folder to Clasnip. The folder shall contain several sub-folders holding sequences in FASTA format. Each <strong>sub-folder</strong> is considered as a group, and each <strong>FASTA file</strong> is considered as a sample.</p><details class="admonition is-details"><summary class="admonition-header">What is the FASTA format?</summary><div class="admonition-body"><p>A FASTA file can contain more than one sequences. A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line (defline) is distinguished from the sequence data by a greater-than (&quot;&gt;&quot;) symbol at the beginning. An example sequence in FASTA format is:</p><pre><code class="language-fasta hljs">&gt;Name1 description
GTTGATGGGGTCATTTGAGTTTATGTTAAGGGCCCATAG
CTCAGGCGGTTAGAGTGCACCC
&gt;Name2 description
ATAGTTTTTTTGTTCTAGGGATTTTTTTTTAGAGCAATA</code></pre></div></details><p>The tree view of the compressed folder looks like this:</p><pre><code class="nohighlight hljs">Pectobacteria_database.zip
├── Pectobacterium_actinidiae
│   ├── GCA_000803315.1_ASM80331v1_genomic.fasta
│   ├── GCA_007833835.1_ASM783383v1_genomic.fasta
│   ├── GCA_017426805.1_Pca_genomic.fasta
│   └── GCA_024343335.1_ASM2434333v1_genomic.fasta
├── Pectobacterium_aquaticum
│   ├── GCA_002869945.1_ASM286994v1_genomic.fasta
│   ├── GCA_003382565.3_ASM338256v3_genomic.fasta
├── Pectobacterium_aroidearum
│   ├── GCA_000023605.1_ASM2360v1_genomic.fasta
│   ├── GCA_013449715.1_ASM1344971v1_genomic.fasta
│   ├── GCA_013913605.1_ASM1391360v1_genomic.fasta</code></pre><p>In the example, the database contain three groups, named <code>Pectobacterium_actinidiae</code>, <code>Pectobacterium_aquaticum</code>, and <code>Pectobacterium_aroidearum</code> . Those groups have 4, 2 and 3 samples,respectively.</p><div class="admonition is-info"><header class="admonition-header">Specs of a database file</header><div class="admonition-body"><ul><li><strong>Accepted compressed methods for database file</strong>: <code>zip</code>, <code>tar.gz</code>, <code>tar.bz2</code>, <code>tar.xz</code>, <code>tar.Z</code></li><li><strong>Allowed FASTA file extensions</strong>: <code>fasta</code>, <code>fa</code>, <code>fna</code>, <code>fs</code>. FASTA should not be compressed. </li><li><strong>Nested folders are allowed</strong>: <code>Group_A/Group_A1/Sample</code> is valid, and the group of Sample is automatically converted to <code>Group_A/Group_A1</code></li><li><strong>Database size limit</strong>: 200M, or shown in Clasnip.com.</li></ul></div></div><h2 id="Reference-sequence"><a class="docs-heading-anchor" href="#Reference-sequence">Reference sequence</a><a id="Reference-sequence-1"></a><a class="docs-heading-anchor-permalink" href="#Reference-sequence" title="Permalink"></a></h2><p>After uploading the database file, user need to select a reference sequence from the FASTA files. The reference should allow all other sequences to be mapped to. If any sequence fails to map to the reference, it will not be analyzed. If a sample barely maps to the reference, the classification performance will be compromised.</p><p><img src="../assets/create_db3.png" alt/></p><p>If the samples are too diverse to be mapped using a single reference, we can</p><ul><li>Click &quot;<strong>build pangenome</strong>&quot;. This function is useful when no reference sequence can represent all groups in the database. You can choose one reference sequence, and a pangenome will be built based on the sequence. The unmapped region of other sequences will be appended to the pangenome, and the pangenome will be the reference genome.</li><li>If the previous method does not generate good mapping results, you can also <strong>manually create a pangenome</strong>, and added the pangenome as a normal FASTA file in the database. Select this as reference, and click &quot;<strong>exclude reference in classification</strong>&quot;. This option is useful when the reference is artificial and only used for a better alignment.</li></ul><h3 id="Submit"><a class="docs-heading-anchor" href="#Submit">Submit</a><a id="Submit-1"></a><a class="docs-heading-anchor-permalink" href="#Submit" title="Permalink"></a></h3><p>Congratualtions! You are all set. Now you can click &quot;<strong>SUBMIT</strong>&quot; to create the database. It may take some time depending on sequence lengths. You can go to the report page to see whether the task is finished.</p><p>After the task is finished, you can review the detailed database information by selecting <strong>Menu</strong> (top bar) -&gt; <strong>Database Information</strong>.</p></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../db_info/">« Database Information</a><a class="docs-footer-nextpage" href="../algorithm/">Algorithm »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.8.1 on <span class="colophon-date" title="Friday 21 February 2025 14:29">Friday 21 February 2025</span>. Using Julia version 1.11.3.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
