<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Database Information · Clasnip Documentation</title><meta name="title" content="Database Information · Clasnip Documentation"/><meta property="og:title" content="Database Information · Clasnip Documentation"/><meta property="twitter:title" content="Database Information · Clasnip Documentation"/><meta name="description" content="Documentation for Clasnip Documentation."/><meta property="og:description" content="Documentation for Clasnip Documentation."/><meta property="twitter:description" content="Documentation for Clasnip Documentation."/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../search_index.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">Clasnip Documentation</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../">Introduction</a></li><li><a class="tocitem" href="../new_analysis/">New Analysis</a></li><li><a class="tocitem" href="../reports/">Job Reports</a></li><li class="is-active"><a class="tocitem" href>Database Information</a><ul class="internal"><li><a class="tocitem" href="#Database-information"><span>Database information</span></a></li><li><a class="tocitem" href="#Detailed-classification-performance"><span>Detailed classification performance</span></a></li></ul></li><li><a class="tocitem" href="../create_db/">Create New Database</a></li><li><a class="tocitem" href="../algorithm/">Algorithm</a></li><li><a class="tocitem" href="../changelog/">Change Log</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Database Information</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Database Information</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/cihga39871/Clasnip.com" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/cihga39871/Clasnip.com/blob/main/docs/src/db_info.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="Clasnip-Database-Information"><a class="docs-heading-anchor" href="#Clasnip-Database-Information">Clasnip Database Information</a><a id="Clasnip-Database-Information-1"></a><a class="docs-heading-anchor-permalink" href="#Clasnip-Database-Information" title="Permalink"></a></h1><p>Clasnip performs sample classification based on specialized databases. When creating a database, Clasnip automatically analyzes performance based on the samples in the database. The database information and performance reports are accessible by all users. </p><p>To browse database information and performance reports, go to <strong>Menu</strong> (top bar) -&gt; <strong>Database Information</strong> -&gt; <strong>Choose A Clasnip Database</strong>.</p><h2 id="Database-information"><a class="docs-heading-anchor" href="#Database-information">Database information</a><a id="Database-information-1"></a><a class="docs-heading-anchor-permalink" href="#Database-information" title="Permalink"></a></h2><h3 id="Metadata"><a class="docs-heading-anchor" href="#Metadata">Metadata</a><a id="Metadata-1"></a><a class="docs-heading-anchor-permalink" href="#Metadata" title="Permalink"></a></h3><p>Database metadata includes:</p><ul><li><strong>Database ID</strong>: The unique ID of clasnip database.</li><li><strong>Taxonomy</strong>: Ccientific name, and taxonomy level.</li><li><strong>Group By</strong>: What are the samples grouped by? Eg: species, stains, haplotypes.</li><li><strong>Region</strong>: What region/genes are covered?</li><li><strong>Reference Sequence</strong>: The reference sequence used for mapping. If it contains <code>.pan.</code>, it is a pan-genome containing not only the reference sequence, but also unmapped regions from other samples. </li><li><strong>Created By</strong>: A user who creates the database.</li><li><strong>Date Created</strong>: Date in the format of <code>YYYY-MM-DD</code>.</li></ul><h3 id="Classification-groups"><a class="docs-heading-anchor" href="#Classification-groups">Classification groups</a><a id="Classification-groups-1"></a><a class="docs-heading-anchor-permalink" href="#Classification-groups" title="Permalink"></a></h3><p>This section lists classification groups in the database.</p><h3 id="Accuracy"><a class="docs-heading-anchor" href="#Accuracy">Accuracy</a><a id="Accuracy-1"></a><a class="docs-heading-anchor-permalink" href="#Accuracy" title="Permalink"></a></h3><p>The percent of accuracy computed based on the samples in the database.</p><p class="math-container">\[Accuracy = \frac{TP + TN}{TP + TN + FP + FN}\]</p><p>where</p><ul><li><strong>TP</strong> (True Positive) = Group X sample correctly identified as X.</li><li><strong>TN</strong> (True Negative) = Non group X sample correctly identified as not X.</li><li><strong>FP</strong> (False Positive) = Non group X sample incorrectly identified as X.</li><li><strong>FN</strong> (False Negative) = Group X sample incorrectly identified as not X.</li></ul><h2 id="Detailed-classification-performance"><a class="docs-heading-anchor" href="#Detailed-classification-performance">Detailed classification performance</a><a id="Detailed-classification-performance-1"></a><a class="docs-heading-anchor-permalink" href="#Detailed-classification-performance" title="Permalink"></a></h2><h3 id="Heatmaps-and-dendrograms"><a class="docs-heading-anchor" href="#Heatmaps-and-dendrograms">Heatmaps and dendrograms</a><a id="Heatmaps-and-dendrograms-1"></a><a class="docs-heading-anchor-permalink" href="#Heatmaps-and-dendrograms" title="Permalink"></a></h3><p>Several plots were generated using different data in the forms of heatmap and dendrogram. Plots may automatically hided, you need to click &quot;<strong>Dendrogram</strong>&quot;, or the sub-buttons under <strong>Heatmap</strong> to view the plot. </p><p>The plots are interactive, and you can</p><ul><li><strong>resize the plot</strong> by dragging the bottom-right corner of the canvas.</li><li><strong>hover</strong> on heatmap cells to show the description of X and Y axes, and numbers. </li><li><strong>zoom</strong> or <strong>pan</strong> plots using mouse.</li><li><strong>quick double click</strong> for auto scale.</li></ul><div class="admonition is-warning"><header class="admonition-header">Warning</header><div class="admonition-body"><p>Some plots might be very big and take a while to show and resize.</p></div></div><h4 id="Group-Similarity-Based-on-Sample-Identities"><a class="docs-heading-anchor" href="#Group-Similarity-Based-on-Sample-Identities">Group Similarity Based on Sample Identities</a><a id="Group-Similarity-Based-on-Sample-Identities-1"></a><a class="docs-heading-anchor-permalink" href="#Group-Similarity-Based-on-Sample-Identities" title="Permalink"></a></h4><p><img src="../assets/gsbsi-heatmap.png" alt/></p><p>The number in the cell is the mean group identity among samples belong to labeled group. X and Y axes all represent groups, but they are a little bit difference. Y axis is the labeled group (true classification), and X axis is the group shown in classification report.</p><h4 id="Group-Similarity-Based-on-SNP-Differences"><a class="docs-heading-anchor" href="#Group-Similarity-Based-on-SNP-Differences">Group Similarity Based on SNP Differences</a><a id="Group-Similarity-Based-on-SNP-Differences-1"></a><a class="docs-heading-anchor-permalink" href="#Group-Similarity-Based-on-SNP-Differences" title="Permalink"></a></h4><p><img src="../assets/gsbsd-heatmap.png" alt/></p><p>The cells are colored with the percent of diverse SNP scores (dark) over covered SNP scores (grey) among two groups. </p><div class="admonition is-info"><header class="admonition-header">The term, SNP, used in Clasnip</header><div class="admonition-body"><p>Clasnip treats SNP, insert and deletion in the same way, and does not distinguish them when computing SNP scores. So the term &quot;SNP&quot; used in Clasnip includes both SNP and INDEL.</p></div></div><h4 id="Sample-Similarity-Based-on-Classification"><a class="docs-heading-anchor" href="#Sample-Similarity-Based-on-Classification">Sample Similarity Based on Classification</a><a id="Sample-Similarity-Based-on-Classification-1"></a><a class="docs-heading-anchor-permalink" href="#Sample-Similarity-Based-on-Classification" title="Permalink"></a></h4><p><img src="../assets/ssbc-heatmap.png" alt/></p><blockquote><p>The Y axis of the figure above is partially displayed.</p></blockquote><p>The heatmap shows the cluster of samples (Y axis) and groups (X axis) in the database. Samples are colored by the classification provided by the database submitter. Cells are colored by identities (percent matched). </p><h4 id="Identity-Quantile-Statistics-for-Each-Group"><a class="docs-heading-anchor" href="#Identity-Quantile-Statistics-for-Each-Group">Identity Quantile Statistics for Each Group</a><a id="Identity-Quantile-Statistics-for-Each-Group-1"></a><a class="docs-heading-anchor-permalink" href="#Identity-Quantile-Statistics-for-Each-Group" title="Permalink"></a></h4><table><tr><th style="text-align: right">#</th><th style="text-align: right">LABELED GROUP</th><th style="text-align: right">N SAMPLE</th><th style="text-align: right">N ACCURATE</th><th style="text-align: right">IDENTITY Q5</th><th style="text-align: right">IDENTITY Q25</th><th style="text-align: right">IDENTITY MEDIAN</th><th style="text-align: right">TRUE POSITIVE RATE</th></tr><tr><td style="text-align: right">1</td><td style="text-align: right">C</td><td style="text-align: right">16</td><td style="text-align: right">16</td><td style="text-align: right">0.864</td><td style="text-align: right">0.892</td><td style="text-align: right">0.929</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">2</td><td style="text-align: right">Chile3</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">3</td><td style="text-align: right">N</td><td style="text-align: right">31</td><td style="text-align: right">31</td><td style="text-align: right">0.93</td><td style="text-align: right">0.969</td><td style="text-align: right">0.985</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">4</td><td style="text-align: right">NTN</td><td style="text-align: right">73</td><td style="text-align: right">73</td><td style="text-align: right">0.957</td><td style="text-align: right">0.969</td><td style="text-align: right">0.991</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">5</td><td style="text-align: right">N_O</td><td style="text-align: right">37</td><td style="text-align: right">37</td><td style="text-align: right">0.954</td><td style="text-align: right">0.984</td><td style="text-align: right">0.987</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">6</td><td style="text-align: right">O</td><td style="text-align: right">87</td><td style="text-align: right">87</td><td style="text-align: right">0.956</td><td style="text-align: right">0.977</td><td style="text-align: right">0.982</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">7</td><td style="text-align: right">Poha</td><td style="text-align: right">6</td><td style="text-align: right">6</td><td style="text-align: right">0.972</td><td style="text-align: right">0.982</td><td style="text-align: right">0.99</td><td style="text-align: right">1</td></tr></table><ul><li><p><strong>LABELED GROUP</strong>: the group labeled in database.</p></li><li><p><strong>N SAMPLE</strong>: the total number of samples with SNP coverage no less than 5. If all samples&#39; coverages are less than 5, all samples are used.</p></li><li><p><strong>N ACCURATE</strong>: the number of samples correctly classified as the labeled group.</p></li><li><p><strong>IDENTITY Q5/25</strong>: the 5/25th quantile of samples&#39; identities belongs to the labeled group.</p></li><li><p><strong>IDENTITY MEDIAN</strong>: the 50th quantile of samples&#39; identities belongs to the labeled group.</p></li><li><p><strong>TRUE POSITIVE RATE (SENSITIVITY)</strong>: the rate of N ACCURATE to N SAMPLE.</p></li></ul><h4 id="Classification-Performance"><a class="docs-heading-anchor" href="#Classification-Performance">Classification Performance</a><a id="Classification-Performance-1"></a><a class="docs-heading-anchor-permalink" href="#Classification-Performance" title="Permalink"></a></h4><table><tr><th style="text-align: right">#</th><th style="text-align: right">LABELED GROUP</th><th style="text-align: right">TPR</th><th style="text-align: right">TNR</th><th style="text-align: right">PPV</th><th style="text-align: right">NPV</th><th style="text-align: right">FNR</th><th style="text-align: right">FPR</th><th style="text-align: right">FDR</th><th style="text-align: right">FOR</th><th style="text-align: right">ACC</th><th style="text-align: right">F1</th></tr><tr><td style="text-align: right">1</td><td style="text-align: right">C</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">2</td><td style="text-align: right">Chile3</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">3</td><td style="text-align: right">N</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">4</td><td style="text-align: right">NTN</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">5</td><td style="text-align: right">N_O</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">6</td><td style="text-align: right">O</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr><tr><td style="text-align: right">7</td><td style="text-align: right">Poha</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">1</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">0</td><td style="text-align: right">1</td><td style="text-align: right">1</td></tr></table><p><a href="https://en.wikipedia.org/wiki/Sensitivity_and_specificity">The terminology and derivations can be found at this page.</a></p><ul><li><p><strong>TP (True Positive)</strong>: a test result that correctly indicates the presence of a condition or characteristic.</p></li><li><p><strong>TN (True Negative)</strong>: a test result that correctly indicates the absence of a condition or characteristic.</p></li><li><p><strong>FP (False Positive)</strong>: a test result which wrongly indicates that a particular condition or attribute is present.</p></li><li><p><strong>FN (False Negative)</strong>: a test result which wrongly indicates that a particular condition or attribute is absent.</p></li><li><p><strong>TPR (Sensitivity, Recall, Hit Rate, True Positive Rate)</strong>: <span>$TP / (TP + FN)$</span>.</p></li><li><p><strong>TNR (Specificity, Selectivity, True Negative Rate)</strong>: <span>$TN / (TN + FP)$</span>.</p></li><li><p><strong>PPV (Precision, Positive Predictive Value)</strong>: <span>$TP / (TP + FP)$</span>.</p></li><li><p><strong>NPV (Negative Predictive Value)</strong>: <span>$TN / (TN + FN)$</span>.</p></li><li><p><strong>FNR (Miss Rate, False Negative Rate)</strong>: <span>$1 - TPR$</span>.</p></li><li><p><strong>FPR (Fall-out, False Positive Rate)</strong>: <span>$1 - TNR$</span>.</p></li><li><p><strong>FDR (False Discovery Rate)</strong>: <span>$1 - PPV$</span>.</p></li><li><p><strong>FOR (False Omission Rate)</strong>: <span>$1 - NPV$</span>.</p></li><li><p><strong>ACC (Accuracy)</strong>: <span>$(TP + TN) / (TP + TN + FP + FN)$</span>.</p></li><li><p><strong>F1 Score</strong>: the harmonic mean of precision and sensitivity.</p></li></ul><div class="admonition is-info"><header class="admonition-header">Cross-validation</header><div class="admonition-body"><p>In addition, if a database is a <em>single-gene</em> database, or &quot;<em>all samples are assemblies</em>&quot; is selected when creating database, Clasnip will evaluate the database comprehensively using cross-validation.  </p><p>Cross-validation is a technique used to assess a classifier model and test its performance without overfitting. Clasnip uses three replicates of 2-fold cross-validation. In a 2-fold cross-validation, samples are evenly divided into two groups. The first group is used to train the model (training set), and the second group is used to test the model (test set). After recording the performance, it will use the second group as the training set, and the first group as the test set. Clasnip does three replications and summarize the performance of training and test sets. Standard deviation is shown in each cell.  </p><p>Also, if a group has only one sample, this group is excluded in cross-validation.</p></div></div><h4 id="Mis-classified-Samples"><a class="docs-heading-anchor" href="#Mis-classified-Samples">Mis-classified Samples</a><a id="Mis-classified-Samples-1"></a><a class="docs-heading-anchor-permalink" href="#Mis-classified-Samples" title="Permalink"></a></h4><p>This table summarizes mis-classified samples. It is useful when evaluating the database performance and checking what samples were misclassified. Sometimes, a public databases will give a wrong label for a sample, and database maintainers can double check the labels when using other methods.</p><p>The columns of this table is similar to classification results. </p><ul><li><p><strong>GROUP</strong>: the group labeled in database.</p></li><li><p><strong>PERCENT MATCHED</strong>: sequence identity, the ratio of MATCHED SNP SCORE and COVERED SNP SCORE.</p></li><li><p><strong>MATCHED SNP SCORE</strong>: sum of scores of all matched SNPs.</p></li><li><p><strong>COVERED SNP SCORE</strong>: sum of scores of all matched and unmatched SNPs.</p></li><li><p><strong>RANK</strong>: the rank of PERCENT MATCHED for the LABELED GROUP.</p></li><li><p><strong>LABEL</strong>: format in LABELED GROUP/SAMPLE NAME.</p></li><li><p><strong>CDF</strong>: the cumulated density where PERCENT MATCHED falls in which quantile of estimated distribution of LABELED GROUP samples in database.</p></li><li><p><strong>PROBABILITY</strong>: the probability of the sample is classified to LABELED GROUP.</p></li></ul><p>If <strong>COVERED SNP SCORE</strong> is small, it probably means insufficient SNPs were covered by your query sequences, so <strong>PERCENT MATCHED</strong> may be stochastic and not predicted precisely.</p><p>The classification accuracy is based on the provided samples and labels, and not guaranteed by Clasnip.</p><h4 id="Low-Coverage-Samples"><a class="docs-heading-anchor" href="#Low-Coverage-Samples">Low Coverage Samples</a><a id="Low-Coverage-Samples-1"></a><a class="docs-heading-anchor-permalink" href="#Low-Coverage-Samples" title="Permalink"></a></h4><p>This table shows samples with low coverages (eg: <strong>COVERED SNP SCORE</strong> &lt; 5). A sample has a low coverage usually means it is unable to successfully map to the reference sequence. If unintentionally, database maintainers may use a better reference sequence, or select &quot;<strong>build pangenome</strong>&quot;, or manually provide a pangenome when building a Clasnip database.</p></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../reports/">« Job Reports</a><a class="docs-footer-nextpage" href="../create_db/">Create New Database »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.8.1 on <span class="colophon-date" title="Friday 21 February 2025 18:51">Friday 21 February 2025</span>. Using Julia version 1.11.3.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
